Issue # 83 (output files)

LTrestka opened the pull request:
- Pass a /path/to/file.ext
  - Creates directory if needed
    - Returns failure user doesn't have permission
  - Writes output to file if provided, otherwise prints it in cli

* closes #83

LTrestka committed:
- Added --output flag
  - Pass a /path/to/file.ext
  - Creates directory if needed
    - Returns failure user doesn't have permission
  - Writes output to file if provided, otherwise prints it in cli

LTrestka committed:
Added sys.exit calls in handle_output, and checks if output flag is a directory path, rather than a standalone file name before trying to create a directory.

LTrestka committed:
- Reversed the checks as requested
- Dropped the else
- Handles errors via a function, prints the error message, and the output response if one occurs

LTrestka committed:
Dropped unused get_output_args() function

shreyb reviewed (changes requested):
My only objection is really with the sys.exits as the way to halt execution of `handle_output`.  I'd rather we either return or raise an Exception, as that's clearer IMO.

LTrestka committed:
Made requested changes:
  - removed excess sys.exit calls in favor of return statements
  - raise exception in handle output if needed, rather than print + sys.exit

shreyb reviewed (commented):
I like this.  I have a couple of minor comments, but that's up to you if you want to implement them.

LTrestka committed:
- Handled edge case where an exception during handle_output with the quiet flag would appear to exit successfully, but a failure had in fact occurred.

- Made --quiet and --dryrun mutually exclusive.

shreyb reviewed (changes requested):
I think there's a typo (`IOError` should be `OSError` in one of the `handle_output` cases). Aside from that (or if I'm wrong), approved.

LTrestka committed:
fixed typo
